src/beta_inc.jl

```diff
@@ -0,0 +1,945 @@
+    t = (1.0/a)^2.0
+    return @horner(t, .833333333333333E-01, -.277777777760991E-02, .793650666825390E-03, -.595202931351870E-03, .837308034031215E-03, -.165322962780713E-02)/a + w
+end
+
+"""
+    esum(mu,x)
+
+Compute ``e^{mu+x}``
+"""
+function esum(mu::Float64, x::Float64)
+    if x > 0.0
+        if mu > 0.0 || mu + x < 0.0
+            w = mu
+            return exp(w)*exp(x)
+        else
+            return exp(mu + x)
+        end
+    elseif mu < 0.0 || mu + x > 0.0
+        w = mu
+        return exp(w)*exp(x)
+    else
+        return exp(mu + x)
+    end
+end
+
+"""
+    brcmp1(mu,a,b,x,y,case)
+
+If case == 1: compute ``e^{mu} * x^{a}y^{b}/B(a,b)``
+else : compute above with mu = 0.0
+"""
+function brcmp1(mu::Float64, a::Float64, b::Float64, x::Float64, y::Float64, case::Bool)
+    a0 = min(a,b)
+    if a0 >= 8.0
+      @goto l100
+    elseif x > 0.375
+      @goto l10
+    end
+    lnx = log(x)
+    lny = log1p(-x)
+    @goto l20
+
+    @label l10
+     if y > 0.375
+      @goto l11
+     end
+     lnx  = log1p(-y)
+     lny = log(y)
+     @goto l20
+   
+    @label l11
+     lnx = log(x)
+     lny = log(y)
+    @label l20
+     z = a*lnx + b*lny
+     if a0 < 1.0
+       @goto l30
+     end
+     z -= logbeta(a,b)
+     return case ? esum(mu,z) : exp(z)
+
+   #PROCEDURE A < 1 OR B < 1
+    @label l30
+     b0 = max(a,b)
+     if b0 >= 8.0
+        @goto l80
+     elseif b0 > 1.0
+        @goto l60
+     end
+
+     # FOR B0 <= 1
+     ans = case ? esum(mu,z) : exp(z)
+     if ans == 0.0
+        return 0.0
+     end
+     apb = a + b
+     if apb > 1.0
+        @goto l40
+     else
+        z = 1.0 + SpecialFunctions.rgamma1pm1(apb)
+        @goto l50
+     end
+    @label l40
+     u = a + b - 1.0
+     z = (1.0 + SpecialFunctions.rgamma1pm1(u))/apb
+
+    @label l50
+     c = (1.0 + SpecialFunctions.rgamma1pm1(a))*(1.0 + SpecialFunctions.rgamma1pm1(b))/z
+     return ans*(a0*c)/(1.0 + a0/b0)
+
+    #  FOR 1 < B0 < 8
+    @label l60
+     u = SpecialFunctions.loggamma1p(a0)
+     n = b0 - 1.0
+     if n < 1
+        @goto l70
+     end
+     c = 1.0
+     for i = 1:n
+        b0 -= 1.0
+        c *= (b0/(a0+b0))
+     end
+     u += log(c)
+
+    @label l70
+     z -= u
+     b0 -= 1.0
+     apb = a0 + b0
+     if apb > 1.0
+        @goto l71
+     else
+        t = 1.0 + SpecialFunctions.rgamma1pm1(apb)
+        @goto l72
+     end
+    @label l71
+     u = a0 + b0 - 1.0
+     t = (1.0 + SpecialFunctions.rgamma1pm1(u))/apb
+    @label l72
+     return a0*(case ? esum(mu,z) : exp(z))*(1.0 + SpecialFunctions.rgamma1pm1(b0))/t
+    
+    # FOR B0 >= 8
+    @label l80
+     u = SpecialFunctions.loggamma1p(a0) + loggammadiv(a0,b0)
+     return a0*(case ? esum(mu, z-u) : exp(z-u))
+    
+    # FOR A >= 8 AND B >= 8
+    @label l100
+     if a > b 
+         @goto l101
+     else
+         h = a/b
+         x0 = h/(1.0 + h)
+         y0 = 1.0/(1.0 + h)
+         lambda = a - (a+b)*x
+         @goto l110
+     end
+    @label l101
```

Comment:
since the above block doesn't flow through, this can be moved to the `@goto l101`.

this should also make the below `@label l110` unnecessary as well.